src/util/ConfigManager/ConfigManager.h

```diff
@@ -0,0 +1,247 @@
+  the configuration option.
+  @param optionName The identifier of the `ConfigOption`.
+  */
+  void verifyPathToConfigOption(const VectorOfKeysForJson& pathToOption,
+                                std::string_view optionName) const;
+
+  /*
+  @brief Adds a configuration option, that can be accessed with the given path.
+
+  @param pathToOption Describes a path in json, that points to the value held by
+  the configuration option. NOTE: The first key must be a string, and the last
+  key must be string, equal to the name of the configuration option.
+  */
+  void addConfigOption(const VectorOfKeysForJson& pathToOption,
+                       ConfigOption&& option);
+
+  // For testing.
+  FRIEND_TEST(ConfigManagerTest, GetConfigurationOptionByNestedKeysTest);
+  FRIEND_TEST(ConfigManagerTest, ParseConfig);
+  FRIEND_TEST(ConfigManagerTest, ParseConfigExceptionTest);
+  FRIEND_TEST(ConfigManagerTest, ParseShortHandTest);
+
+ public:
+  /*
+  @brief Creates and adds a new configuration option.
+
+  @tparam OptionType The type of value, the configuration option can hold.
+
+  @param pathToOption Describes a path in json, that points to the value held by
+  the configuration option. The last key in the vector is the name of the
+  configuration option. NOTE: Both the first, and the last key have to be
+  strings.
+  @param optionDescription A description for the configuration option.
+  @param variableToPutValueOfTheOptionIn The value held by the configuration
+  option will be copied into this variable, whenever the value in the
+  configuration option changes.
+  @param defaultValue A default value for the configuration option. If none is
+  given, signified by an empty optional, then a value for the configuration
+  option MUST be given at runtime.
+  */
+  template <typename OptionType>
+  requires ad_utility::isTypeContainedIn<OptionType,
+                                         ConfigOption::AvailableTypes>
+  void createConfigOption(const VectorOfKeysForJson& pathToOption,
+                          std::string_view optionDescription,
+                          OptionType* variableToPutValueOfTheOptionIn,
+                          std::optional<OptionType> defaultValue =
+                              std::optional<OptionType>(std::nullopt)) {
+    /*
+    In this case, we have an invalid path and can't create a `ConfigOption`.
+    Because we don't want to copy the corresponding exception out of
+    `verifyPathToConfigOption`, that would be to much code duplication, we
+    instead call it in such a way, that it will throw the corresponding
+    exception instead.
+    If we don't have this case, then the actual path check will be done by
+    `addConfigOption`. No reason, to do it double.
+    */
+    if (pathToOption.empty() ||
+        !std::holds_alternative<std::string>(pathToOption.back())) {
+      verifyPathToConfigOption(pathToOption, "");
+    }
+
```

Comment:
Okay, no I understand the problem here. But would the following work:
call
```
verifyPathToConfigOption(...);
AD_CORRECTNESS_CHECK(!pathToOption.empty());
```

here, and then don't check the path again in the `addConfigOption` funktion (it is private anyway).

The AD_CORRECTNESS_CHECK is there to automatically document that the `verify..` function does the right thing (otherwise the call to `.back()` is unsafe.